src/DwhManager.php

```diff
@@ -0,0 +1,414 @@
+    {
+        $grantedRoles = $this->checker->getGrantedRolesOfRole($roleName);
+        $toBeRevoked = array_diff($grantedRoles, $desiredRoles);
+        $toBeGranted = array_diff($desiredRoles, $grantedRoles);
+        foreach ($toBeRevoked as $revokedRole) {
+            $this->connection->revokeRoleGrantFromRole($revokedRole, $roleName);
+            $this->logger->info(sprintf(
+                'Role "%s" has been revoked from role "%s"',
+                $revokedRole,
+                $roleName
+            ));
+        }
+        foreach ($toBeGranted as $grantedRole) {
+            $this->connection->grantRoleToRole($grantedRole, $roleName);
+            $this->logger->info(sprintf(
+                'Role "%s" has been granted to role "%s"',
+                $grantedRole,
+                $roleName
+            ));
+        }
+    }
+
+    private function ensureSchemaExists(string $schemaName): void
+    {
+        if (!$this->checker->existsSchema($schemaName)) {
+            $this->connection->createSchema($schemaName);
+            $this->logger->info(sprintf(
+                'Created schema "%s"',
+                $schemaName
+            ));
+        } else {
+            $this->logger->info(sprintf(
+                'Schema "%s" exists',
+                $schemaName
+            ));
+        }
+    }
+
+    private function ensureUserExists(string $userName, array $options): void
+    {
+        if (!$this->checker->existsUser($userName)) {
+            $password = $this->generatePassword();
+            $this->connection->createUser(
+                $userName,
+                $password,
+                $options
+            );
+            if (isset($options['login_name'])) {
+                $this->logger->info(sprintf(
+                    'Created user "%s" (%s) with password "%s"',
+                    $options['login_name'],
+                    $userName,
+                    $password
+                ));
+            } else {
+                $this->logger->info(sprintf(
+                    'Created user "%s" with password "%s"',
+                    $userName,
+                    $password
+                ));
+            }
+        } else {
+            unset($options['login_name']);
+            $this->connection->alterUser(
+                $userName,
+                $options
+            );
+            $this->logger->info(sprintf(
+                'User "%s" already exists',
+                $userName
+            ));
+        }
+    }
+
+    private function generatePassword(): string
+    {
+        $randomLibFactory = new Factory();
+        $password = $randomLibFactory
+            ->getMediumStrengthGenerator()
+            ->generateString(self::GENERATED_PASSWORD_LENGTH);
+        return $password;
+    }
+
+    private function getOwnSchemaNameFromUser(User $user): string
+    {
+        $schemaName = $this->sanitizeAsIdentifier($user->getEmail()) . '_schema_rw';
```

Comment:
nevyhodíme to `_schema` z toho suffixu? To že jde o schema  je jasný z kontextu a imho to bude vypadat trochu lépe v UI.
![image](https://user-images.githubusercontent.com/903531/41702928-01c74010-7532-11e8-8d1f-f1fbd12dceaa.png)